src/main/java/org/dependencytrack/resources/v1/ComponentResource.java

```diff
@@ -118,7 +120,11 @@ public Response getComponentByUuid(
             if (component != null) {
                 final Project project = component.getProject();
                 if (qm.hasAccess(super.getPrincipal(), project)) {
+                    final PackageURL purl = component.getPurl();
+                    final RepositoryMetaComponent metaComponent = qm.getRepositoryMetaComponent(
+                            RepositoryType.resolve(purl), purl.getNamespace(), purl.getName());
```

Comment:
That will raise a NPE when `purl` is `null`. Not every component has a PURL, some may have CPE, SWID tag or none of them at all. There should be a test to cover this scenario as well.